### PR TITLE
waiting for changes request

### PR DESCRIPTION
**Pull request by MaksymKos**
https://maksymkos.github.io/js_employees_table_DOM/

### CODE REVIEW

File: src/scripts/main.js [hunk 1]
```diff
@@ -1,3 +1,352 @@
...
+            'error'
+          );
+          break;
+        }
+
+        td.innerText = input.value;
+        break;
+
+      case 3:
+        if (input.value < 18 || input.value > 90) {
```
Comment (Viktor-Kost) [inline src/scripts/main.js:326]: create constants for such values as 18 or 90 to avoid using `magic` numbers

File: src/scripts/main.js [hunk 2]
```diff
@@ -1,3 +1,352 @@
...
+      input.type = 'number';
+      input.value = td.innerHTML;
+      break;
+
+    case 4:
+      input = document.createElement('input');
+      input.type = 'number';
+      input.placeholder = 'just a number';
+      input.value = Number(td.innerHTML.slice(1).split(',').join(''));
+      break;
```
Comment (Viktor-Kost) [inline src/scripts/main.js:287]: add default behavior

File: src/scripts/main.js [hunk 3]
```diff
@@ -1,3 +1,352 @@
...
+  `;
+
+  table.append(tr);
+}
+
+//
+//
+//
+// fourth task(notification)
+function formValidation(data) {
```
Comment (Viktor-Kost) [inline src/scripts/main.js:169]: function names should start from a verb

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]
Nice!)
Some moments can be improved

File: src/scripts/main.js (hunk 2 above)
Comment (MaksymKos) [reply src/scripts/main.js:287]: can u explain?

File: src/scripts/main.js [hunk 4]
```diff
@@ -1,3 +1,335 @@
...
+    <td>${data.office}</td>
+    <td>${data.age}</td>
+    <td>$${Number(data.salary).toLocaleString('en-US')}</td>
+  `;
+
+  table.append(tr);
+}
+
+const minAge = 18;
+const maxAge = 90;
```
Comment (Viktor-Kost) [inline src/scripts/main.js:151]: usually such constant we write like this:
```suggestion
const MIN_AGE = 18;
const MAX_AGE = 90;
```

File: src/scripts/main.js [hunk 5]
```diff
@@ -1,3 +1,335 @@
...
+    data: td.innerHTML,
+  };
+
+  let input;
+
+  switch (index) {
+    case 0:
+      input = document.createElement('input');
+      input.type = 'text';
+      input.placeholder = 'at least 4 character';
```
Comment (Viktor-Kost) [inline src/scripts/main.js:232]: 1) move `4` to a constant
```
const MIN_CHARACTERS = 4
```
2) use for template string
```suggestion
      input.placeholder = `at least ${MIN_CHARACTERS} characters`;
```

File: src/scripts/main.js [hunk 6]
```diff
@@ -1,3 +1,335 @@
...
+  input.onblur = function() {
+    finishTdEdit(td, input, index);
+  };
+}
+
+function finishTdEdit(td, input, index) {
+  if (input.value !== '') {
+    switch (index) {
+      case 0:
+        if (input.value.length < 4) {
```
Comment (Viktor-Kost) [inline src/scripts/main.js:294]: reuse constant
```suggestion
        if (input.value.length < MIN_CHARACTERS) {
```

File: src/scripts/main.js [hunk 7]
```diff
@@ -1,3 +1,335 @@
...
+          td.innerText = editingTd.data;
+
+          pushNotification(
+            'Error',
+            'Name must contain more then 3 character',
```
Comment (Viktor-Kost) [inline src/scripts/main.js:299]: reuse constant
```suggestion
            `Name must contain at least ${MIN_CHARACTERS} characters`,
```

File: src/scripts/main.js [hunk 8]
```diff
@@ -1,3 +1,335 @@
...
+        td.innerText = input.value;
+        break;
+
+      case 3:
+        if (input.value < minAge || input.value > maxAge) {
+          td.innerText = editingTd.data;
+
+          pushNotification(
+            'Error',
+            'Age must be between 18 and 90',
```
Comment (Viktor-Kost) [inline src/scripts/main.js:314]: reuse constants
```suggestion
            `Age must be between ${MIN_AGE} and ${MAX_AGE}`,
```

File: src/scripts/main.js [hunk 9]
```diff
@@ -1,3 +1,335 @@
...
+    <td>$${Number(data.salary).toLocaleString('en-US')}</td>
+  `;
+
+  table.append(tr);
+}
+
+const minAge = 18;
+const maxAge = 90;
+
+function validationsOfInformation(data) {
```
Comment (Viktor-Kost) [inline src/scripts/main.js:153]: function with a verb.
As an example: 
```suggestion
function checkIsValid(data) {
```

File: src/scripts/main.js [hunk 10]
```diff
@@ -1,3 +1,335 @@
...
+  if (data.name.length < 4) {
+    pushNotification(
+      'Error',
+      'Name must contain more then 3 character',
```
Comment (Viktor-Kost) [inline src/scripts/main.js:157]: reuse constants

File: src/scripts/main.js [hunk 11]
```diff
@@ -1,3 +1,335 @@
...
+      'error'
+    );
+
+    return false;
+  }
+
+  if (data.age < minAge || data.age > maxAge) {
+    pushNotification(
+      'Error',
+      'Age must be between 18 and 90',
```
Comment (Viktor-Kost) [inline src/scripts/main.js:167]: reuse constants

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]
Much better)
some minor fixes needed

**Review by YegorKorenkov — APPROVED** [review]
Looks good

**Pull request closed without merging**